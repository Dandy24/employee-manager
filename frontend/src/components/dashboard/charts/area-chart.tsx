import { observer } from 'mobx-react-lite';
import Title from 'antd/lib/typography/Title';
import { Area, AreaChart, CartesianGrid, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts';
import { Col } from 'antd';
import React from 'react';
import { GraphDataInterface } from '../../../models/interfaces/graph-data-interface';
import moment from 'moment';

export interface AreaChartProps {
    title: string;
    data: GraphDataInterface[];
    dataKey1: string;
    dataName1: string;
    xAxisKey: string;
}

export const MyAreaChart: React.FC<AreaChartProps> = observer((props: AreaChartProps) => {
    const { title, data, dataKey1, dataName1, xAxisKey } = props;

    return (
        <Col style={{ width: '50%', height: '100%' }}>
            <Title level={2} style={{ textAlign: 'center' }}>
                {title}
            </Title>
            <ResponsiveContainer width="100%" height="90%">
                <AreaChart
                    width={500}
                    height={400}
                    data={data}
                    margin={{
                        top: 10,
                        right: 30,
                        left: 0,
                        bottom: 0,
                    }}
                >
                    <CartesianGrid strokeDasharray="4 4" />
                    <XAxis
                        dataKey={xAxisKey}
                        tickMargin={10}
                        tickFormatter={(value) => moment(value, 'YYYY-MM-DD').format('MMMM YY')}
                    />
                    <YAxis />
                    <Tooltip
                        formatter={(value) => `${value}%`}
                        labelFormatter={(label: string) => <p>{moment(label, 'YYYY-MM-DD').format('MMMM YY')}</p>}
                    />
                    <Area type="monotone" dataKey={dataKey1} name={dataName1} stroke="#8884d8" fill="#8884d8" />
                </AreaChart>
            </ResponsiveContainer>
        </Col>
    );
});
