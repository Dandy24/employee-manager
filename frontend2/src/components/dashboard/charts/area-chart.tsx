import { observer } from 'mobx-react-lite';
import Title from 'antd/lib/typography/Title';
import { Area, AreaChart, CartesianGrid, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts';
import { Col } from 'antd';
import React from 'react';

export interface AreaChartProps {
    title: string;
    data: any;
    dataKey1: string;
    dataName1: string;
    xAxisKey: string;
}

export const MyAreaChart: React.FC<AreaChartProps> = observer((props: AreaChartProps) => {
    const { title, data, dataKey1, dataName1, xAxisKey } = props;

    return (
        <Col style={{ width: '50%', height: '100%' }}>
            <Title level={2} style={{ textAlign: 'center' }}>
                {title}
            </Title>
            <ResponsiveContainer width="100%" height="90%">
                <AreaChart
                    width={500}
                    height={400}
                    data={data}
                    margin={{
                        top: 10,
                        right: 30,
                        left: 0,
                        bottom: 0,
                    }}
                >
                    <CartesianGrid strokeDasharray="3 3" />
                    <XAxis dataKey={xAxisKey} />
                    <YAxis />
                    <Tooltip formatter={(value) => `${value}%`} />
                    <Area type="monotone" dataKey={dataKey1} name={dataName1} stroke="#8884d8" fill="#8884d8" />
                </AreaChart>
            </ResponsiveContainer>
        </Col>
    );
});
