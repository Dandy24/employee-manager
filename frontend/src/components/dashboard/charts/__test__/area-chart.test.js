import { render } from '@testing-library/react';
import React from 'react';
import { MyAreaChart } from '../area-chart';

jest.mock('recharts', () => {
    const OriginalModule = jest.requireActual('recharts');

    return {
        ...OriginalModule,
        ResponsiveContainer: ({ height, children }) => (
            <OriginalModule.ResponsiveContainer width={800} height={500}>
                {children}
            </OriginalModule.ResponsiveContainer>
        ),
    };
});

test('Area chart matches snapshot', async () => {
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation((query) => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // deprecated
            removeListener: jest.fn(), // deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });

    const testData = [
        { name: '2022-03-01', value: 62 },
        { name: '2022-02-01', value: 77 },
    ];

    const view = render(
        <MyAreaChart
            xAxisKey="name"
            dataKey1="value"
            dataName1="Efektivita"
            title="VÃ½voj efektivity"
            data={testData}
        />,
    );

    // eslint-disable-next-line testing-library/no-node-access,testing-library/no-container
    const areaChart = view.container.querySelector('#area-chart');

    expect(areaChart).toMatchSnapshot();
});
