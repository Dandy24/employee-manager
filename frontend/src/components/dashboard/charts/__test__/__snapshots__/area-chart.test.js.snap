// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area chart matches snapshot 1`] = `
<svg
  class="recharts-surface"
  height="500"
  id="area-chart"
  version="1.1"
  viewBox="0 0 800 500"
  width="800"
>
  <defs>
    <clippath
      id="area-chart-clip"
    >
      <rect
        height="460"
        width="710"
        x="60"
        y="10"
      />
    </clippath>
  </defs>
  <g
    class="recharts-cartesian-grid"
  >
    <g
      class="recharts-cartesian-grid-horizontal"
    >
      <line
        fill="none"
        height="460"
        offset="[object Object]"
        stroke="#ccc"
        stroke-dasharray="4 4"
        width="710"
        x="60"
        x1="60"
        x2="770"
        y="10"
        y1="470"
        y2="470"
      />
      <line
        fill="none"
        height="460"
        offset="[object Object]"
        stroke="#ccc"
        stroke-dasharray="4 4"
        width="710"
        x="60"
        x1="60"
        x2="770"
        y="10"
        y1="355"
        y2="355"
      />
      <line
        fill="none"
        height="460"
        offset="[object Object]"
        stroke="#ccc"
        stroke-dasharray="4 4"
        width="710"
        x="60"
        x1="60"
        x2="770"
        y="10"
        y1="240"
        y2="240"
      />
      <line
        fill="none"
        height="460"
        offset="[object Object]"
        stroke="#ccc"
        stroke-dasharray="4 4"
        width="710"
        x="60"
        x1="60"
        x2="770"
        y="10"
        y1="125"
        y2="125"
      />
      <line
        fill="none"
        height="460"
        offset="[object Object]"
        stroke="#ccc"
        stroke-dasharray="4 4"
        width="710"
        x="60"
        x1="60"
        x2="770"
        y="10"
        y1="10"
        y2="10"
      />
    </g>
    <g
      class="recharts-cartesian-grid-vertical"
    >
      <line
        fill="none"
        height="460"
        offset="[object Object]"
        stroke="#ccc"
        stroke-dasharray="4 4"
        width="710"
        x="60"
        x1="60"
        x2="60"
        y="10"
        y1="10"
        y2="470"
      />
      <line
        fill="none"
        height="460"
        offset="[object Object]"
        stroke="#ccc"
        stroke-dasharray="4 4"
        width="710"
        x="60"
        x1="770"
        x2="770"
        y="10"
        y1="10"
        y2="470"
      />
    </g>
  </g>
  <g
    class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
  >
    <line
      class="recharts-cartesian-axis-line"
      fill="none"
      height="30"
      orientation="bottom"
      stroke="#666"
      type="category"
      width="710"
      x="60"
      x1="60"
      x2="770"
      y="470"
      y1="470"
      y2="470"
    />
    <g
      class="recharts-cartesian-axis-ticks"
    >
      <g
        class="recharts-layer recharts-cartesian-axis-tick"
      >
        <line
          class="recharts-cartesian-axis-tick-line"
          fill="none"
          height="30"
          orientation="bottom"
          stroke="#666"
          type="category"
          width="710"
          x="60"
          x1="60"
          x2="60"
          y="470"
          y1="476"
          y2="470"
        />
        <text
          class="recharts-text recharts-cartesian-axis-tick-value"
          fill="#666"
          height="30"
          orientation="bottom"
          stroke="none"
          text-anchor="middle"
          type="category"
          width="710"
          x="60"
          y="486"
        >
          <tspan
            dy="0.71em"
            x="60"
          >
            March 22
          </tspan>
        </text>
      </g>
      <g
        class="recharts-layer recharts-cartesian-axis-tick"
      >
        <line
          class="recharts-cartesian-axis-tick-line"
          fill="none"
          height="30"
          orientation="bottom"
          stroke="#666"
          type="category"
          width="710"
          x="60"
          x1="770"
          x2="770"
          y="470"
          y1="476"
          y2="470"
        />
        <text
          class="recharts-text recharts-cartesian-axis-tick-value"
          fill="#666"
          height="30"
          orientation="bottom"
          stroke="none"
          text-anchor="middle"
          type="category"
          width="710"
          x="770"
          y="486"
        >
          <tspan
            dy="0.71em"
            x="770"
          >
            February 22
          </tspan>
        </text>
      </g>
    </g>
  </g>
  <g
    class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
  >
    <line
      class="recharts-cartesian-axis-line"
      fill="none"
      height="460"
      orientation="left"
      stroke="#666"
      type="number"
      width="60"
      x="0"
      x1="60"
      x2="60"
      y="10"
      y1="10"
      y2="470"
    />
    <g
      class="recharts-cartesian-axis-ticks"
    >
      <g
        class="recharts-layer recharts-cartesian-axis-tick"
      >
        <line
          class="recharts-cartesian-axis-tick-line"
          fill="none"
          height="460"
          orientation="left"
          stroke="#666"
          type="number"
          width="60"
          x="0"
          x1="54"
          x2="60"
          y="10"
          y1="470"
          y2="470"
        />
        <text
          class="recharts-text recharts-cartesian-axis-tick-value"
          fill="#666"
          height="460"
          orientation="left"
          stroke="none"
          text-anchor="end"
          type="number"
          width="60"
          x="52"
          y="470"
        >
          <tspan
            dy="0.355em"
            x="52"
          >
            0
          </tspan>
        </text>
      </g>
      <g
        class="recharts-layer recharts-cartesian-axis-tick"
      >
        <line
          class="recharts-cartesian-axis-tick-line"
          fill="none"
          height="460"
          orientation="left"
          stroke="#666"
          type="number"
          width="60"
          x="0"
          x1="54"
          x2="60"
          y="10"
          y1="355"
          y2="355"
        />
        <text
          class="recharts-text recharts-cartesian-axis-tick-value"
          fill="#666"
          height="460"
          orientation="left"
          stroke="none"
          text-anchor="end"
          type="number"
          width="60"
          x="52"
          y="355"
        >
          <tspan
            dy="0.355em"
            x="52"
          >
            20
          </tspan>
        </text>
      </g>
      <g
        class="recharts-layer recharts-cartesian-axis-tick"
      >
        <line
          class="recharts-cartesian-axis-tick-line"
          fill="none"
          height="460"
          orientation="left"
          stroke="#666"
          type="number"
          width="60"
          x="0"
          x1="54"
          x2="60"
          y="10"
          y1="240"
          y2="240"
        />
        <text
          class="recharts-text recharts-cartesian-axis-tick-value"
          fill="#666"
          height="460"
          orientation="left"
          stroke="none"
          text-anchor="end"
          type="number"
          width="60"
          x="52"
          y="240"
        >
          <tspan
            dy="0.355em"
            x="52"
          >
            40
          </tspan>
        </text>
      </g>
      <g
        class="recharts-layer recharts-cartesian-axis-tick"
      >
        <line
          class="recharts-cartesian-axis-tick-line"
          fill="none"
          height="460"
          orientation="left"
          stroke="#666"
          type="number"
          width="60"
          x="0"
          x1="54"
          x2="60"
          y="10"
          y1="125"
          y2="125"
        />
        <text
          class="recharts-text recharts-cartesian-axis-tick-value"
          fill="#666"
          height="460"
          orientation="left"
          stroke="none"
          text-anchor="end"
          type="number"
          width="60"
          x="52"
          y="125"
        >
          <tspan
            dy="0.355em"
            x="52"
          >
            60
          </tspan>
        </text>
      </g>
      <g
        class="recharts-layer recharts-cartesian-axis-tick"
      >
        <line
          class="recharts-cartesian-axis-tick-line"
          fill="none"
          height="460"
          orientation="left"
          stroke="#666"
          type="number"
          width="60"
          x="0"
          x1="54"
          x2="60"
          y="10"
          y1="10"
          y2="10"
        />
        <text
          class="recharts-text recharts-cartesian-axis-tick-value"
          fill="#666"
          height="460"
          orientation="left"
          stroke="none"
          text-anchor="end"
          type="number"
          width="60"
          x="52"
          y="10"
        >
          <tspan
            dy="0.355em"
            x="52"
          >
            80
          </tspan>
        </text>
      </g>
    </g>
  </g>
  <g
    class="recharts-layer recharts-area"
  >
    <g
      class="recharts-layer"
    >
      <defs>
        <clippath
          id="animationClipPath-recharts-area-2"
        >
          <rect
            height="471"
            width="0"
            x="60"
            y="0"
          />
        </clippath>
      </defs>
      <g
        class="recharts-layer"
        clip-path="url(#animationClipPath-recharts-area-2)"
      >
        <g
          class="recharts-layer"
        >
          <path
            class="recharts-curve recharts-area-area"
            d="M60,113.49999999999999L770,27.24999999999999L770,470L60,470Z"
            fill="#8884d8"
            fill-opacity="0.6"
            height="460"
            name="Efektivita"
            points="[object Object],[object Object]"
            stroke="none"
            type="monotone"
            width="710"
          />
          <path
            class="recharts-curve recharts-area-curve"
            d="M60,113.49999999999999L770,27.24999999999999"
            fill="none"
            fill-opacity="0.6"
            height="460"
            name="Efektivita"
            points="[object Object],[object Object]"
            stroke="#8884d8"
            type="monotone"
            width="710"
          />
        </g>
      </g>
    </g>
  </g>
</svg>
`;
